sprout/core/codegen/generate_dataclasses.py

```diff
@@ -0,0 +1,71 @@
+import subprocess
+from pathlib import Path
+
+from sprout.core.fetch_json_from_url import fetch_json_from_url
+from sprout.core.write_json import write_json
+
+SCHEMA_URL = "https://datapackage.org/profiles/2.0/datapackage.json"
+CODEGEN = Path("sprout", "core", "codegen")
+SCHEMA_PATH = CODEGEN / "schema.json"
+OUTPUT_PATH = CODEGEN / "generated_properties.py"
+CUSTOM_TEMPLATE_DIR = CODEGEN
+CUSTOM_FORMATTERS_MODULE = f"{".".join(CODEGEN.parts)}.custom_formatters"
+
+
+def run_codegen():
+    """Uses datamodel-codegen to extract dataclasses from the specified JSON schema.
+
+    The output is post-processed using custom templating and a custom formatter.
+    """
+    subprocess.run([
+        "datamodel-codegen",
+        "--input", SCHEMA_PATH,
+        "--input-file-type", "jsonschema",
+        "--output", OUTPUT_PATH,
+        "--output-model-type", "dataclasses.dataclass",
+        "--custom-template-dir", CUSTOM_TEMPLATE_DIR,
+        "--custom-formatters", CUSTOM_FORMATTERS_MODULE,
+        "--custom-file-header", "# ruff: noqa",
```

Comment:
```suggestion
        "--custom-file-header", "# ruff: noqa\n# NOTE: This content was autogenerated, no need to edit it.",
```

Just to have a disclaimer somewhere about where the file came from.